packages/runtime/src/runtime/customWidget.tsx

```diff
@@ -39,20 +39,21 @@ const OnLoadAction: React.FC<
     context: Record<string, unknown>;
   }>
 > = ({ action, children, context }) => {
-  const onLoadAction = useEnsembleAction(action);
+  const onLoadActionRef = useRef(useEnsembleAction(action));
```

Comment:
This is a sign you're breaking the render cycle in some way - why did you need this?

Reply:
because custom widget's onLoad is firing twice :(

Reply:
Like we discussed, if it's not obvious what the issue is, let's leave this implementation as is for now and do a follow up PR.